### PR TITLE
Move ESQL yaml tests to core

### PR DESCRIPTION
**Pull request by dnhatn**
This PR moves ESQL YAML tests to the core so that we can run them with different modules (e.g., the mixed cluster tests) and even with serverless in the future. YAML tests for other plugins are also in the core.

### CODE REVIEW

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/80_text.yml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ---
 setup:
   - skip:
+      version: " - 7.10.99"
+      reason: "ESQL is available in 7.11+"
```
Comment (alex-spies) [inline x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/80_text.yml:5]: Heya, is that a typo? I would think we need to `s/7/8/g`, applies to all `.yml` files.

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_enrich.yml [hunk 2]
```diff
@@ -1,6 +1,8 @@
 ---
 setup:
   - skip:
+      version: " - 7.10.99"
+      reason: "ESQL is available in 7.11+"
```
Comment (ChrisHegarty) [inline x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_enrich.yml:5]: Is this correct? Or should it be 8.11? ( I could be missing how all this actually fits together )

**Comment by dnhatn** [comment]
Apologies for the mistake. I've pushed the fix. The incorrect skip versions didn't cause any issues since we don't have a mixed cluster in x-pack (though I can add one later). These skips are essentially no-op. I will enable these tests in the ESQL mixed cluster module in a follow-up.

**Review by alex-spies — APPROVED** [review]
LGTM now!

**Review by ChrisHegarty — APPROVED** [review]
LGTM

**Comment by dnhatn** [comment]
@alex-spies @ChrisHegarty Thanks for reviews.

**Pull request merged**